src/mappings/remark.ts

```diff
@@ -3,49 +3,113 @@ import { CallHandlerContext } from '../processor'
 import events from '../types/events'
 import { config } from '../config'
 import { TokenTransfer, Transfer } from '../model'
+import { Call } from '@subsquid/substrate-processor'
 
-export async function handleSystemRemark(ctx: CallHandlerContext) {
+export async function handleBatchWithRemark(ctx: CallHandlerContext) {
     let origin = codec(config.prefix).encode(ctx.call.origin.value.value)
 
-    let matchingBalanceTransfer = await ctx.store.findOne(Transfer, {
-        where: {
-            from: origin,
-            blockNumber: ctx.call.block.height,
-        },
-    })
-
-    let matchingTokenTransfer = await ctx.store.findOne(TokenTransfer, {
-        where: {
-            from: origin,
-            blockNumber: ctx.call.block.height,
-        },
-    })
-
-    // Error if found more than one on the union of both token and balances, or none
-    if (matchingBalanceTransfer && matchingTokenTransfer) {
-        ctx.log.info(
-            `Error when finding the correct transfer for remark with id ${ctx.call.id}. Inconsistent number of transfers for user: ${origin}`
-        )
+    // filter out batch calls where there one transfer and
+    // one remark only
+    if (!hasSystemWithRemark(ctx.call)) {
         return
     }
+    const calls = ctx.call.args.calls
+    const kinds = calls.map((call: any) => call.__kind)
+    if (kinds.length != 2) return
```

Comment:
We can consider lifting this restriction to `kinds.length < 2` so that we can also consider batch calls that for some reason contain more than a transfer and a remark. WDYT @gianfra-t?

Reply:
I think alternatively we could accept batch and batchAll with many calls, as long as there is one remark and one of the transfers. Let me know if that is okay.